djangocms_versioning/admin.py

```diff
@@ -563,13 +563,15 @@ def nr(self, obj):
     def content_link(self, obj):
         """Display html for the content preview url"""
         content = obj.content
-        url = get_preview_url(content)
-
-        return format_html(
-            '<a target="_top" class="js-versioning-close-sideframe" href="{url}">{label}</a>',
-            url=url,
-            label=content,
-        )
+        language = content.language if hasattr(content, 'language') else get_language()
+        with override(language):
+            url = get_preview_url(content)
```

Comment:
```suggestion
            url = get_preview_url(content, language=language)
```
When passing the language argument to `get_preview_url` you can skip the `with` block.